test/unit/test_label_procs.py

```diff
@@ -0,0 +1,186 @@
+from common_utils import run_proc
+from common_utils import validate_row_count
+
+
+def test_smoke_create_drop_label(conn_cnx, timestamp_string):
+    label = generate_unique_name("label", timestamp_string)
+    sql = f"call ADMIN.CREATE_LABEL('{label}', NULL, NULL, 'rows_produced > 100');"
+
+    # create_label returns NULL in case of successful label creation
+    assert (
+        run_proc(conn_cnx, sql)[0] is None
+    ), "Stored procedure did not return NULL value!"
+
+    # make sure it was created with correct properties
+    sql = f"""select count(*) from INTERNAL.labels where
+                 name = '{label}' and
+                 group_name is null and
+                 group_rank is null and
+                 label_created_at is not null and
+                 condition = 'rows_produced > 100' and
+                 enabled is null
+        """
+    assert validate_row_count(conn_cnx, sql)[0] == 1, "Label was not found!"
+
+    # drop label
+    sql = f"call ADMIN.DELETE_LABEL('{label}');"
+    assert (
+        run_proc(conn_cnx, sql)[0] == "done"
+    ), "Stored procedure did not return NULL value!"
+
+
+def test_smoke_update_label(conn_cnx, timestamp_string):
+    label = generate_unique_name("label", timestamp_string)
+    sql = f"call ADMIN.CREATE_LABEL('{label}', NULL, NULL, 'rows_produced > 100');"
+
+    # create_label returns NULL in case of successful label creation
+    assert (
+        run_proc(conn_cnx, sql)[0] is None
+    ), "Stored procedure did not return NULL value!"
+
+    # make sure it was created with correct properties
+    sql = f"""select count(*) from INTERNAL.labels where
+                name = '{label}' and
+                group_name is null and
+                group_rank is null and
+                label_created_at is not null and
+                condition = 'rows_produced > 100' and
+                enabled is null
+        """
+    assert validate_row_count(conn_cnx, sql)[0] == 1, "Label was not found!"
+
+    # update label
+    sql = f"call ADMIN.UPDATE_LABEL('{label}', '{label}', NULL, NULL, 'compilation_time > 3000');"
+    assert (
+        run_proc(conn_cnx, sql)[0] is None
+    ), "Stored procedure did not return NULL value!"
+
+    # make sure it was created with correct properties
+    sql = f"""select count(*) from INTERNAL.labels where
+                name = '{label}' and
+                group_name is null and
+                group_rank is null and
+                label_created_at is not null and
+                condition = 'compilation_time > 3000' and
+                enabled is null
+        """
+    assert validate_row_count(conn_cnx, sql)[0] == 1, "Label was not found!"
+
+
+# List of test cases with statements and expected error messages
+test_cases = [
+    (
+        "call ADMIN.CREATE_LABEL(NULL, NULL, NULL, 'compilation_time > 5000');",
+        "Name must not be null.",
+    ),
+    (
+        "call ADMIN.CREATE_LABEL('{label}', NULL, 100, 'compilation_time > 5000');",
```

Comment:
Is an empty label name (`''`) allowed or disallowed?

Reply:
great catch! Added test case.